utils/src/main/java/io/atomix/utils/serializer/KryoIOPool.java

```diff
@@ -0,0 +1,35 @@
+package io.atomix.utils.serializer;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Function;
+
+class KryoIOPool<T> {
```

Comment:
only "risks" I see here is that the pool queue is unbounded and we may retain elements with large bufferSizes.

Reply:
bytearrayoutputstreams with buffer exceeding MAX_BUFFER_SIZE are no longer retained in the pool.

Reply:
@kuujo maybe we could wrap the pool objects with a SoftReference like we already use for kryopool to keep it consistent?